### ISSUE TITLE
Add playlist position media variable

### ISSUE THREAD

**Issue by Funtime60** [meeb/tubesync#463]
Would it be possible to add a media variable such as {Playlist_Number} so external tools could better organize my playlist? I'm trying to convert my old batch scripts I used to run manually to an automatic system and this would be really helpful. Thanks.

**Comment by meeb** [comment]
Unfortunately it's not easily possible. This is discussed at length in some other now closed tickets. There's some effort to have a general arbitrary incremental number available as a variable, but this isn't specifically the index of the video on a playlist.

**Comment by Makhuta** [comment]
Would it be possible to use the position in the playlist for the variable? Since it is already used in for example making the .nfo. Also this is how I understood the question of @Funtime60 , if I am wrong sorry for intervening but the resolve I described I would really appreciate

**Comment by Funtime60** [comment]
I'm not sure what you're suggesting? But also I don't think I ended up using this tool. In the end, I wrote my own tool to manage yt-dlp and do what my scripts used to.

**Comment by meeb** [comment]
The .nfo file "position" counter is basically just an arbitrary counter when the .nfo file is generated, it's not useful for a variable used for filenames which are written incrementally, the counter would repeat itself.

**Comment by Makhuta** [comment]
It wouldn't be "counter" since when you are downloading the playlist from YT the yt-dl (usually) provides you with video position in playlist.
Let's take the playlist in the [example](https://www.youtube.com/playlist?list=PL590L5WQmH8dpP0RyH5pCfIaDEdt9nk7r) when adding source from playlist

![image](https://github.com/user-attachments/assets/ad3ad7fe-e92e-4792-a7fc-a806d5562855)

each video from the playlist has it position in playlist which is most likely used in .nfo episode number (probably also in JSON haven't really tried)

```nfo
<?xml version='1.0' encoding='utf8'?>
<episodedetails>
  <title>První výzkumy DINOSAURŮ!  [JurskýPark] #3</title>
  <showtitle>Gejmr &amp; Jirka - Jurský park </showtitle>
  <season>1</season>
  <episode>3</episode>
  <ratings>
    <rating name="youtube" max="5" default="True">
      <value>None</value>
      <votes>24778</votes>
    </rating>
  </ratings>
  <plot>Instagram: https://www.instagram.com/GEJMR
Facebook: http://www.facebook.com/GEJMRcz
Twitter: https://twitter.com/GEJMRcz
WEB: http://www.GEJMR.eu/

Jirka: https://www.youtube.com/user/jirkakral

Business email: kontaktgejmr@gmail.com

Intro vectors by Freepik</plot>
  <thumb>Gejmr &amp; Jirka - Jurský park 🌱🐊 - EgVMJgFS-01c - První výzkumy DINOSAURŮ! 💻 [JurskýPark] #3.jpg</thumb>
  <mpaa>0</mpaa>
  <runtime>1532</runtime>
  <id>gVMJgFS-01c</id>
  <uniqueid type="youtube" default="True">gVMJgFS-01c</uniqueid>
  <studio>GEJMR</studio>
  <aired>2018-07-07</aired>
  <dateadded>2024-12-04 22:46:26</dateadded>
  <genre>Gaming</genre>
</episodedetails>
```
^^^^ this is example of video from another playlist which was 3rd in the playlist. As you can see the episode is 3 in this .nfo and I was wondering if this could be used in variable for naming

**Comment by meeb** [comment]
Yes, this value is usually available (it's sometimes not). The issue here is that tubesync doesn't do one-off downloads and playlists change over time. For example, a video is added at the start of a playlist. Now every existing previous item already downloaded by tubesync has its position off by one. If this position variable is used in the filename they probably need to also all be renamed, and all the metadata rewritten, nfo files updated, etc.

It's only trivial to implement if you download a playlist then never update it again.

**Comment by Makhuta** [comment]
I understand, I am not sure how really you are storing the the videos etc. so I don't know how hard it would be to update the metadata and rename files like you said. In the future I will try to explore the repo to hopefully help. Even tho will it ve possible to do (only the variable based on position in playlist) even if the renaming etc. will be issue but for now it could just be noted that it could make this problem.

PS. When I think about it since the position is already used in .nfo metadata, and I think it isn't updated when the position in playlist change (haven't confirmed that so it might not be the case) so it wouldn't be that much of a problem

**Comment by meeb** [comment]
It's by no means impossible, it's just quite a lot of work that no-one has done yet to keep it all in sync and rename everything etc. And yes currently .nfo files aren't rewritten at the moment.

**Comment by Makhuta** [comment]
I personally would really appreciate it even without the syncing so I can make playlist of for example my favourite YouTuber series and download it like TV series formated similary to Sonarr. (as I said the syncing isn't even necesarry since like you said would be really hard to do)

**Comment by tcely** [comment]
@meeb Does #586 help with the renumbering problems at all?

**Comment by tcely** [comment]
The changes for this were merged in #617.

**Comment by meeb** [comment]
Thanks @tcely ! This seems resolved so I'll close it for now. Please open a new issue if you have further suggestions.
